src/packet/pacer.rs

```diff
@@ -563,6 +526,59 @@ impl LeakyBucketPacer {
             .min(self.padding_bitrate * MAX_DEBT_IN_TIME);
         crate::packet::bwe::macros::log_pacer_padding_debt!(self.padding_debt.as_bytes_usize());
     }
+
+    /// Optimistically attempt to create a padding request.
+    ///
+    /// Returns `Some(PaddingRequest)` if padding is enabled and the current queue state
+    /// allows padding, otherwise returns `None`.
+    fn maybe_create_padding_request(&self) -> Option<PaddingRequest> {
+        // We must have no debt.
+        if self.media_debt != DataSize::ZERO || self.padding_debt != DataSize::ZERO {
+            return None;
+        }
+
+        // Queues must be empty.
+        let all_queues_empty = self
+            .queue_states
+            .iter()
+            .all(|q| q.snapshot.packet_count == 0);
+        if !all_queues_empty {
+            return None;
+        }
+
+        // We must have sent some media.
+        if self.last_send_time.is_none() {
```

Comment:
I dunno about clippy here. It suggests

```rust
        // We must have sent some media.
        self.last_send_time?;
```

which just seems harder to read, I'm inclined to ignore it

Reply:
I think it's ok with `?` – one always needs to be on the lookout for them,.